server/chat-commands.js

```diff
@@ -3796,6 +3796,49 @@ const commands = {
 	},
 	importinputloghelp: [`/importinputlog [inputlog] - Starts a battle with a given inputlog. Requires: + % @ & ~`],
 
+	acceptdraw: 'accepttie',
+	offerdraw: 'offertie',
+	accepttie: 'offertie',
+	offertie(target, room, user, connection, cmd) {
+		const battle = room.battle;
+		if (!battle) return this.errorReply("Must be in a battle room.");
+		if (!Config.allowrequestingties) {
+			return this.errorReply("This server does not allow offering ties.");
+		}
+		if (!this.can('roomvoice', null, room)) return;
+		if (cmd === 'accepttie' && !battle.players.some(player => player.wantsTie)) {
+			return this.errorReply("No tie offer available. It might have been automatically withdrawn as the current turn started.");
+		}
+		const playerIds = Object.keys(battle.playerTable);
+		if (!battle.players.some(player => player.wantsTie)) {
+			for (const player of playerIds) {
+				if (player === user.userid) {
+					battle.playerTable[player].wantsTie = true;
+					continue;
+				}
+				Users(player).sendTo(
+					room,
+					Chat.html`|html|${user.name} wants this game to end in a tie; <button class="button" name="send" value="/accepttie">accept tie</button>?`
+				);
+			}
+			this.add(`${user.name} is offering a tie`);
```

Comment:
Maybe move this line above the others, so it appears above the "accept tie" button.

Reply:
Actually, once this is above the others, you don't even need to repeat it in "accept tie" button.

Remember to `room.update()` if you need to.